### PR TITLE
Improve log messages in SkipDocumentFilter and RegexFilter

### PR DESCRIPTION
**Pull request by BrettMichaelJohnson**
Fix b/38359039 SkipDocumentFilter should log "No transmission decision"
instead of "Not Skipping"

Fix b/38356739 Param and metadata names are case-sensitive in transforms

### CODE REVIEW

**Review by PapaJoltOmega — APPROVED** [review]
LGTM

File: src/com/google/enterprise/adaptor/prebuilt/RegexFilter.java [hunk 1]
```diff
@@ -185,7 +186,15 @@ private RegexFilter(String key, Keyset keyset, Pattern pattern,
    */
   private boolean foundInMetadata(Metadata metadata) {
     boolean found = false;
-    for (String value : metadata.getAllValues(key)) {
+    Set<String> values = metadata.getAllValues(key);
+    if (values.isEmpty()) {
+      if (metadata.getKeys().contains(key)) {
+        log.finest("No values for key `" + key + "' in metadata.");
```
Comment (wiarlawd) [inline src/com/google/enterprise/adaptor/prebuilt/RegexFilter.java]: Include docid in these messages, and use MessageFormat style.

**Review by wiarlawd — CHANGES_REQUESTED** [review]
This is not a fix for b/38356739, it just adds some logging suggested in that bug. I would move the other bug into the summary and add these an an "other change" in the description.

File: src/com/google/enterprise/adaptor/prebuilt/SkipDocumentFilter.java [hunk 2]
```diff
@@ -142,7 +144,15 @@ private SkipDocumentFilter(String propertyName, Pattern pattern,
    */
   private boolean foundInMetadata(Metadata metadata) {
     boolean found = false;
-    for (String value : metadata.getAllValues(propertyName)) {
+    Set<String> values = metadata.getAllValues(propertyName);
+    if (values.isEmpty() && log.isLoggable(Level.FINEST)) {
+      if (metadata.getKeys().contains(propertyName)) {
+        log.finest("No values for key `" + propertyName + "' in metadata.");
```
Comment (wiarlawd) [inline src/com/google/enterprise/adaptor/prebuilt/SkipDocumentFilter.java]: The use of ` feels like an affectation, and is used only 8 times in the full googlegsa code base (compared to 141 occurrences of '"). Let's not add 6 more. If we takeout the existing two each in SkipDocumentFilter and RegexDocumentFilter, then the library would have zero.

File: src/com/google/enterprise/adaptor/prebuilt/SkipDocumentFilter.java [hunk 3]
```diff
@@ -162,6 +172,8 @@ private boolean foundInParams(Map<String, String> params) {
     boolean found = false;
     if (params.containsKey(propertyName)) {
       found = pattern.matcher(params.get(propertyName)).find();
+    } else if (log.isLoggable(Level.FINEST)) {
```
Comment (wiarlawd) [inline src/com/google/enterprise/adaptor/prebuilt/SkipDocumentFilter.java]: Let's use MessageFormat-style on this one.

File: src/com/google/enterprise/adaptor/prebuilt/RegexFilter.java [hunk 4]
```diff
@@ -205,9 +214,11 @@ private boolean foundInParams(Map<String, String> params) {
     boolean found = false;
     if (params.containsKey(key)) {
       found = pattern.matcher(params.get(key)).find();
+    } else if (log.isLoggable(Level.FINEST)) {
```
Comment (wiarlawd) [inline src/com/google/enterprise/adaptor/prebuilt/RegexFilter.java]: The isLoggable check is redundant here.

**Review by wiarlawd — APPROVED** [review]
LGTM, with a nit.

File: src/com/google/enterprise/adaptor/prebuilt/RegexFilter.java (hunk 4 above)
Comment (BrettMichaelJohnson) [reply src/com/google/enterprise/adaptor/prebuilt/RegexFilter.java]: Done.

**Pull request merged**